api/envoy/api/v2/route/route.proto

```diff
@@ -757,6 +757,22 @@ message RouteAction {
   // This can be used to prevent unexpected upstream request timeouts due to potentially long
   // time gaps between gRPC request and response in gRPC streaming mode.
   google.protobuf.Duration max_grpc_timeout = 23 [(gogoproto.stdduration) = true];
+
+  // Allows enabling and disabling upgrades on a per-route basis.
+  // This overrides any enabled/disabled upgrade filter chain specified in the
+  // HttpConnectionManager
+  // :ref:upgrade_configs`
+  // <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.upgrade_configs>`
+  // but does not affect any custom filter chain specified there.
+  message UpgradeConfig {
+    // The case-insensitive name of this upgrade, e.g. "websocket".
```

Comment:
What would be an invalid string?  I don't think we do http compliance checking on any of our header strings thusfar

Reply:
I'm mostly thinking of manually edited control planes that might deliver routes with spelling mistake. i.e someone disables `wesocket` [sic] upgrades, and they get unexpected results because of the spelling error.

Reply:
Yeah, I don't think we can do much about typos unfortunately, and we can't cross-check against HCM because you can do per-route only